downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/HumanPathwaysWithDiagrams.java

```diff
@@ -0,0 +1,56 @@
+package org.reactome.release.downloadDirectory;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Collection;
+
+public class HumanPathwaysWithDiagrams {
+
+    private static final String humanPathwaysWithDiagramsFilename = "humanPathwaysWithDiagrams.txt";
+    private static final Integer HUMAN_DATABASE_ID = 48887;
+
+    /**
+     * This class goes through all Human Pathway instances checking if it has a PathwayDiagram associated with it.
+     * If it does, it is recorded in a file containing the Pathways db id, name and if it has a disease instance associated with it.
+     * @param dba -- MySQLAdaptor
+     * @param releaseNumber -- String, release number
+     * @throws Exception
+     */
+    public static void execute(MySQLAdaptor dba, String releaseNumber) throws Exception {
+
+        // Create new file
+        File humanPathwaysWithDiagramsFile = new File(humanPathwaysWithDiagramsFilename);
+        humanPathwaysWithDiagramsFile.delete();
+        humanPathwaysWithDiagramsFile.createNewFile();
+
+        // Fetch all Human Pathway instances
+        Collection<GKInstance> humanPathways = dba.fetchInstanceByAttribute(ReactomeJavaConstants.Pathway, ReactomeJavaConstants.species, "=", HUMAN_DATABASE_ID);
+
+        for (GKInstance humanPathway : humanPathways) {
+            // Default disease value is false
+            boolean isDisease = false;
+
+            // If returned value from 'disease' query is not null, its considered to be a disease-containing Pathway
+            if (humanPathway.getAttributeValue(ReactomeJavaConstants.disease) != null) {
+                isDisease = true;
+            }
```

Comment:
This could be a ternary or a simple private method and called in place of the temp variable on line 48.
```
private boolean isDisease(GKInstance pathway) throws Exception {
     return humanPathway.getAttributeValue(ReactomeJavaConstants.disease) != null;
}
```